lib/client/twofa/twofa.go

```diff
@@ -225,22 +226,36 @@ func authenticateUser(
 	}
 	// upgrade to u2f
 	successful2fa := false
+	useNewLib := true
 	if !skip2fa {
 		if allowU2F {
-			devices, err := u2fhid.Devices()
-			if err != nil {
-				logger.Fatal(err)
-				return err
-			}
-			if len(devices) > 0 {
-
-				err = u2f.DoU2FAuthenticate(
+			if useNewLib {
+				//err = u2f.WithDevicesDoU2FAuthenticate(u2fhost.Devices(),
+				//	client, baseUrl, userAgentString, logger)
+				err = u2f.WithDevicesDoWebAuthnAuthenticate(u2fhost.Devices(),
 					client, baseUrl, userAgentString, logger)
 				if err != nil {
-
+					logger.Fatal(err)
 					return err
 				}
 				successful2fa = true
+
+			} else {
+				devices, err := u2fhid.Devices()
+				if err != nil {
+					logger.Fatal(err)
+					return err
```

Comment:
There should be no code after a call to `Fatal()`. I suggest removing the call to `Fatal()`.

Reply:
done